File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int max = 0;
        for(int i : tangerine) {
            max = Math.max(i,max);
        }
        int[] categories = new int[max+1];
        for(int i = 0; i < tangerine.length; i++) {
            categories[tangerine[i]]++;
        }
        Arrays.sort(categories);
        int sum = 0;
        for(int i = categories.length-1; i >=0; i--) {
            sum += categories[i];
            answer++;
            if (sum >= k) {
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    const obj = {};
    for (const t of tangerine) {
        obj[t] = (obj[t] || 0) + 1;
    }
    const arr = [...Object.values(obj)].sort((a,b) => b - a);
    for (const element of arr) {
        k -= element;
        answer++;
        if (k <= 0) break;
    }
    return answer;
}